### PR TITLE
Feature/issue 13

### LINKED ISSUE #13: Adicionar opção para parametrizar url
Criar opção para a busca no mock-server utilizar variáveis de url

Ex:

http://api/users/:userId

Dessa forma qualquer chamado retorne dados:
http://api/users/1
http://api/users/10
http://api/users/58123456789

### LINKED ISSUE #26: Criar mecanismo para encontrar local requests similares
Criar mecanismo para encontrar local requests similares

Ex:
https://swapi.co/api/people/1
https://swapi.co/api/people/5
https://swapi.co/api/people/15

Utilizando o cache only deve encontrar se existir ao menos uma request cadastrada
Ex:
https://swapi.co/api/people/:id

### PR DESCRIPTION
**Pull request by LucasdeCastro**
**Description**
This PR resolve the #13 issue
Close #13 
Close #26

### CODE REVIEW

**Comment by LucasdeCastro** [comment]
This PR resolve the issue #26

File: middlewares/cached.js [hunk 1]
```diff
@@ -4,17 +4,53 @@ const fetch = require("node-fetch");
...
+    const likeSorted = like.list.sort((a, b) => {
+      if (a.url > b.url) return 1;
+      if (a.url === b.url) return 0;
+      return -1;
+    });
+
+    if (likeSorted.length > 0) {
+      res.append("x-request-mock", "true");
+      res.status(likeSorted[0].status);
+      res.send(likeSorted[0].response);
```
Comment (rafaellucio) [inline middlewares/cached.js]: In this point, you received a list sorted with one or many items, but you respond only first item. Why?

Comment (LucasdeCastro) [reply middlewares/cached.js]: likeSorted is the list of suggestions and was ordenade by the url size so the first item is the most recommended suggestion

Comment (rafaellucio) [reply middlewares/cached.js]: Your likeSorted method can return first item. Do you can use unshift array method?

File: middlewares/cached.js [hunk 2]
```diff
@@ -4,17 +4,58 @@ const fetch = require("node-fetch");
...
 const path = require("path");
+const URL = require("url");
 const { getConfig } = require("../config");
 
+const getLikeURL = (acc, char, key, url) => {
+  const str = acc.str + char;
+  const list = acc.list.filter(({ url }) => url.startsWith(str));
+  if (list.length === 0) {
+    if (url.length / 2 <= str.length) return acc;
+    return { list: [], str };
```
Comment (iamtchelo) [inline middlewares/cached.js]: Se `list.length === 0`, logo ele já é um `[]`, então não precisa desse `return { list: [], str };`

**Pull request merged**